expr/builtins/url_email.go

```diff
@@ -39,6 +39,17 @@ func emailEval(ctx expr.EvalContext, args []value.Value) (value.Value, bool) {
 	if val == nil || val.Nil() || val.Err() {
 		return nil, false
 	}
+
+	if s, ok := val.(value.Slice); ok {
+		out := make([]string, 0, s.Len())
+		for _, sv := range s.SliceValue() {
+			if v, ok := emailEval(ctx, append([]value.Value{sv}, args[1:]...)); ok {
+				out = append(out, v.ToString())
+			}
+		}
+		return value.NewStringsValue(out), len(out) > 0
```

Comment:
this returns a different type than the [`Type` method defines](https://github.com/lytics/qlbridge/pull/26/files#diff-a07768616bfbf98cc67588d4c82ad6f0eddde020c1fb61ab2c5287872671644bR28) (which is `value.StringType`) and will affect expression validation.

Reply:
yeah, we're already doing the same thing for assert.Format/Length... 🤔 
In theory that only breaks expression validation if some function were wrapping this that applied validation that `args[0]` was of `value.StringType`... so this SHOULD allow for all normal operation of validation, while ALSO allowing the existing "assertion" functions to be applied as simple wrappings of the existing expression (as long as no "assertion" function REQUIRES value.StringType input, in which would could potentially get around that by forcing order of application 🤷🏻 )